Purchases/Public/Purchases.swift

```diff
@@ -775,8 +788,15 @@ public extension Purchases {
     @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
     @objc(purchasePackage:withDiscount:completion:)
     func purchase(package: Package, discount: SKPaymentDiscount, completion: @escaping PurchaseCompletedBlock) {
-        let payment = storeKitWrapper.payment(withProduct: package.product, discount: discount)
-        purchase(product: package.product,
+        // todo: add support for SK2 with discounts, move to new class
+        // https://github.com/RevenueCat/purchases-ios/issues/848
+        guard let sk1ProductDetails = package.productDetails as? SK1ProductDetails else {
```

Comment:
do we plan to eventually pull any sk1 vs sk2 logic out somewhere? i.e. into purchases orchestrator or elsewhere?

Reply:
yeah, there should be a single place that decides whether to use sk1 or sk2 implementations

Reply:
https://app.shortcut.com/revenuecat/story/11122/feature-flag-shouldusesk2whenavailable